server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/SetMessagesMethodStepdefs.java

```diff
@@ -216,8 +215,10 @@ public void createDraft(String username, String message) throws Throwable {
                 "  ]" +
                 "]");
             mainStepdefs.awaitMethod.run();
-            String messageId = httpClient.jsonPath.read("[0][1].created." + message + ".id");
-            getMessagesMethodStepdefs.addMessageId(message, mainStepdefs.messageIdFactory.fromString(messageId));
+            Optional.ofNullable(
+                httpClient.jsonPath.<String>read("[0][1].created." + message + ".id"))
+                .map(mainStepdefs.messageIdFactory::fromString)
+                .ifPresent(id -> getMessagesMethodStepdefs.addMessageId(message, id));
```

Comment:
I don't like test conditions in test. what's the point here?

Reply:
The point is:
 - If the create request succeeded I want to be able to retrieve the id.
 - If the create request fails, then we should not attempt to  read a message that do not exist.

I did this because I thought duplicating a Jerkins step made even less sens.

Reply:
ok in this particular case it makes sense imho. Can you then rename the use case as "try to create a draft message" then?

Reply:
:+1: